packages/components/src/systems/SignatureAttestation/SignatureAttestationModal.tsx

```diff
@@ -0,0 +1,252 @@
+    return await client.mutate<AgreeToSignatureAttestationType, { version: string }>({
+      mutation: AgreeToSignatureAttestation,
+      variables: { version },
+      refetchQueries: ['GetCurrentUserSignatureAttestationStatus'],
+      awaitRefetchQueries: true,
+      update: (cache, { data }) => {
+        if (data?.agreeToSignatureAttestation) {
+          cache.writeQuery({
+            query: GetCurrentUserSignatureAttestationStatus,
+            data: {
+              me: {
+                signatureAttestationStatus: {
+                  __typename: 'CompletedSignatureAttestation',
+                  agreedAt: new Date(),
+                  version
+                }
+              }
+            }
+          });
+        }
+      }
+    });
+  };
+
+export interface SignatureAttestationModalProps {
+  client: ApolloClient<any>;
+  children: JSXElement;
+  onAgreeComplete?: () => void;
+}
+
+type Status =
+  | {
+      status: 'LOADING';
+    }
+  | { status: 'NEEDS ATTESTATION'; version: string }
+  | { status: 'COMPLETE' }
+  | { status: 'ERROR'; errors: any[] };
+
+export const SignatureAttestationModal = (props: SignatureAttestationModalProps) => {
+  let ref: Ref<any> | undefined;
+  const [status, setStatus] = createSignal<Status>({ status: 'LOADING' });
+
+  const refreshAttestationStatus = async () => {
+    const req = await getCurrentUserSignatureAttestationStatus(props.client);
+    if (req.error || req.errors) {
+      setStatus({
+        status: 'ERROR',
+        errors: [
+          ...(req.errors?.map((e) => e.message) ?? []),
+          ...(req.error ? [req.error.message] : [])
+        ]
+      });
+    } else if (!req.data?.me?.signatureAttestationStatus) {
+      setStatus({
+        status: 'ERROR',
+        errors: ['Could not get user data. Please refresh and try again']
+      });
+    } else if (
+      req.data?.me?.signatureAttestationStatus?.__typename === 'CompletedSignatureAttestation' ||
+      req.data?.me?.signatureAttestationStatus?.__typename === 'NotApplicableSignatureAttestation'
+    ) {
+      setStatus({ status: 'COMPLETE' });
+    } else
+      setStatus({
+        status: 'NEEDS ATTESTATION',
+        version: req.data.me.signatureAttestationStatus.version
+      });
+  };
+
+  const dispatchSignatureAttestationAgreed = () => {
+    const event = new CustomEvent('photon-signature-attestation-agreed', {
+      composed: true,
+      bubbles: true,
+      detail: {}
+    });
+    ref?.dispatchEvent(event);
+  };
+
+  const dispatchSignatureAttestationCanceled = () => {
+    const event = new CustomEvent('photon-signature-attestation-canceled', {
+      composed: true,
+      bubbles: true,
+      detail: {}
+    });
+    ref?.dispatchEvent(event);
+  };
+
+  createEffect(() => {
+    if (status().status === 'COMPLETE') {
+      props.onAgreeComplete?.();
+      dispatchSignatureAttestationAgreed();
+    }
+  });
+
+  createEffect(() => {
```

Comment:
is this happening on mount? if so use `onMount`

Reply:
It _should_ happen any time the state of the modal / component changes. I dont understand the lifecycle enough. I think thats more than just onMount. Especially because I dont think you're guaranteed to have the `client` on mount?